File: src/main/java/no/novari/flyt/archive/gateway/resource/configuration/ResourcePublishingPullConfigurationProperties.java
package no.novari.flyt.archive.gateway.resource.configuration;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.time.Duration;

@Getter
@Setter
@ConfigurationProperties(prefix = "novari.flyt.archive.gateway.resource.publishing.pull")
public class ResourcePublishingPullConfigurationProperties {
    private Duration initialDelay;
    private Duration fixedDelay;
}

File: src/main/kotlin/no/novari/flyt/archive/gateway/resource/configuration/ResourcePublishingPullConfigurationProperties.kt
package no.novari.flyt.archive.gateway.resource.configuration

import org.springframework.boot.context.properties.ConfigurationProperties
import java.time.Duration

@ConfigurationProperties(prefix = "novari.flyt.archive.gateway.resource.publishing.pull")
class ResourcePublishingPullConfigurationProperties {
    var initialDelay: Duration? = null
    var fixedDelay: Duration? = null
}
